src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -280,6 +307,12 @@ private ParseUserDefinitionResult ParseUDFsAndNamedFormulas(string script, Parse
 
                         // Parse expression
                         var result = ParseExpr(Precedence.None);
+                        if (result.GetType() == typeof(TypeLiteralNode))

```

Comment:
nit: `result is TypeLiteralNode typeLitNode` instead of .GetType and the cast on 312